packages/utils/upgrade/src/modules/upgrader/upgrader.ts

```diff
@@ -79,7 +79,7 @@ export class Upgrader implements UpgraderInterface {
       `${this.target.major}.${this.target.minor}.${this.target.patch}`
     );
 
-    this.logger?.debug(
+    this.logger?.debug?.(
```

Comment:
Is there a reason we don't require a logger everywhere? and/or if we don't want one we create a stub that doesn't actually log anything?

Reply:
In the future, we might want to programmatically create an upgrader instance that won't log anything (silent).

We could create a stub, but I found it simpler to assume we might not have a logger at all. What do you think?